### ISSUE TITLE
Explicitly bind to 0.0.0.0 

### ISSUE THREAD

**Issue by dsmiley** [docker-solr/docker-solr#318]
In Solr master (9x), Solr no longer listens to 0.0.0.0, it listens to 127.0.0.1.  For a non-containerized Solr, this means only the local machine can communicate to Solr.  But in a containerized environment, this is an additional / redundant barrier with Docker's own networking.  I worked around this today by toggling this env var on the CLI
`-e SOLR_JETTY_HOST="0.0.0.0"`
I propose that this be set automatically, perhaps in docker-entrypoint.sh 

If a user doing local Solr dev wants to ensure that only the local machine can interact with the container, then the port binding can specify the host: `-p 127.0.0.1:8983:8983`  It'd be nice if Docker defaulted that way.

**Comment by Stono** [comment]
Got caught out by this one too; nothing in the changelog about it that I can see.    It confused me a bit as we also use `-Djetty.host=0.0.0.0`, which appears to be ignored, `SOLR_JETTY_HOST` seemed to sort it though (thank you).

Agreed it should probably be defaulted to the previous behaviour (0.0.0.0) and then migrated to as an explicit breaking change.

**Comment by Stono** [comment]
Nope, I lied - it's right here on https://github.com/apache/solr/blob/main/solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc.

**Comment by dsmiley** [comment]
Solr 9 is already setting this in its Dockerfile so I suppose I should close this issue: https://github.com/apache/solr/blob/main/solr/docker/templates/Dockerfile.body.template#L54
